R/formatter.r

```diff
@@ -1,3 +1,72 @@
+#' Number formatter: a generic formatter for numbers
+#'
+#' @return \code{number_format} returns a function with single parameter
+#' \code{x}, a numeric vector, that returns a character vector
+#' @param x a numeric vector to format
+#' @param accuracy number to round to, \code{NULL} for automatic guess
+#' @param scale a scaling factor (useful if the underlying is on another
+#' scale, e.g. for computing percentages or thousands)
+#' @param prefix,suffix symbols to display before and after value
+#' @param big.mark character used between every 3 digits to separate thousands
+#' @param decimal.mark the character to be used to indicate the numeric
+#' decimal point
+#' @param trim logical, if \code{FALSE}, values are right-justified to a common
+#' width (see \code{\link[base]{format}})
+#' @param ... other arguments passed on to \code{\link[base]{format}}
+#' @export
+#' @examples
+#' v <- c(12.3, 4, 12345.789, 0.0002)
+#' number(v)
+#' number(v, big.mark = ",")
+#' number(v, accuracy = .001)
+#' number(v, accuracy = .001, decimal.mark = ",")
+#' number(v, accuracy = .5)
+#'
+#' my_format <- number_format(big.mark = "'", decimal.mark = ",")
+#' my_format(v)
+#'
+#' # Per mille
+#' per_mille <- number_format(scale = 1000,
+#'                            suffix = "\u2030",
+#'                            accuracy = .1)
+#' per_mille(v)
+number_format <- function(accuracy = 1, scale = 1, prefix = "",
```

Comment:
I think it would make more sense to default `big.mark` to `","`, and set `decimal.mark = NULL`

Then we can do:

```R
if (is.null(decimal.mark)) {
  decimal.mark = if (identical(big.mark, ",")) "." else ","
}
```

And then document `big.mark` and `decimal.mark` together.

```R

Reply:
Probably because I'm coming from a different culture (I'm French and comma is used as a decimal separator), I'm personally not in favour of using  a comma as the default thousands separator, for the following reasons:

- using comma as the thousands separator is the purpose of `comma_format` specific formatter;
- in an international perspective, it would be more relevant to use a space as a thousands separator, as used for example by *Lancet* journal (a space for thousands separator and a dot for decimal separator being a good compromise for a good understanding for an international audience);
- since 2003, the use of spaces as separators (for example: 20 000 and 1 000 000 for "twenty thousand" and "one million") has been officially endorsed by SI/ISO 31-0 standard, as well as by the International Bureau of Weights and Measures and the International Union of Pure and Applied Chemistry (IUPAC), the American Medical Association's widely followed AMA Manual of Style, and the Metrication Board, among others. (cf. https://www.wikiwand.com/en/Decimal_separator#/Digit_grouping)

If a space is used for throusands separator, I would suggest to simply state "." as the default value for `decimal.mark`, rather than `NULL` as it avoir to have to consider an implicit rule.

Reply:
OK, I have changed the default behaviour,.

By default, `big.mark` is NULL. If NULL, a comma will be used except if `decimal.mark` is already a comma and in such case a space will be used for thousands separators.